pkg/agent/nodeportlocal/k8s/k8s_controller.go

```diff
@@ -0,0 +1,177 @@
+// +build !windows
+
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"fmt"
+	"hash/fnv"
+	"time"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/nodeportlocal/portcache"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+const (
+	minRetryDelay = 2 * time.Second
+	maxRetryDelay = 120 * time.Second
+	numWorkers    = 4
+)
+
+func bkt(key string, numWorkers uint32) uint32 {
+	h := fnv.New32a()
+	h.Write([]byte(key))
+	bkt := h.Sum32() & (numWorkers - 1)
+	return bkt
+}
+
+type Controller struct {
+	portTable   *portcache.PortTable
+	kubeClient  clientset.Interface
+	queue       workqueue.RateLimitingInterface
+	Ctrl        cache.Controller
+	CacheStore  cache.Store
+	OldObjStore cache.Store
```

Comment:
looking at the code, I am struggling to understand what `OldObjStore` is for. It looks like it should be used to store some realized state, but it doesn't seem to be the case. Looks like you are using it to store the old object passed in handlers, but IMO it goes against the workqueue pattern.